crimsobot/cogs/cringo.py

```diff
@@ -1,225 +1,423 @@
+            title='',
+            descr=f'**{who}** has joined the game!',
+            color_name='green',
+            thumb_name=''
+        )
+
+        return embed
+
+    def generate_mismatch_embed(self) -> discord.Embed:
+        embed = c.crimbed(
+            title=None,
+            descr='Mismatch(es) detected. You lose points for that!',
+            color_name='orange',
+        )
+
+        return embed
+
+    def remove_from_game(self, user: discord.User) -> discord.Embed:
+        embed = c.crimbed(
+            title=None,
+            descr=f'{user} has left the game.',
+            color_name='yellow',
+        )
+
+        return embed
+
+    def get_scoreboard(self, *, game_finished: bool = False) -> CringoScoreboard:
+        scoreboard_rows = []
+        sorted_players = sorted(self.players, key=lambda item: item.score, reverse=True)
+        for player in sorted_players:
+            coin_display = 'zero' if self.cursed else round(player.winnings, 1)
+            if game_finished:
+                row = f'{player.user} · **{player.score}** points · **{coin_display}** coin'
+            else:
+                row = f'{player.user} · **{player.score}** points'
 
-        # send everyone their card
-        for player in list_of_players:
+            scoreboard_rows.append(row)
+
+        return CringoScoreboard('\n'.join(scoreboard_rows), sorted_players[0])
+
+    def prettify_card(self, card: List[List[str]]) -> str:
+        # we add the blank emoji to the first line to accommodate compact mode w/o resizing emojis
+        pretty_string = '\n'.join('\u200A'.join(sublist) for sublist in card)
+        return f'<:blank:589560784485613570>\n{pretty_string}'
+
+    async def send_cards(self) -> None:
+        for player in self.players.copy():
             try:
-                await player.user.send(await cringo.deliver_card(player.card))
+                pretty_card = self.prettify_card(player.card)
+                await player.user.send(pretty_card)
             except discord.errors.Forbidden:
-                embed = await cringo.player_remove(list_of_players, player)
-                await ctx.send(embed)
+                self.players.remove(player)
+                self.cog.all_players.discard(player.user.id)
+                embed = c.crimbed(
+                    title=None,
+                    descr=f'{player.user} has left the game.',
+                    color_name='yellow',
+                )
 
-        # initial game variables
-        turn_timer = CRINGO_RULES['timer'][card_size]
-        turn = 1
-        total_turns = CRINGO_RULES['turns'][card_size]
-        emojis_already_used: List[str] = []
+                await self.context.send(embed)
 
-        while turn <= total_turns and list_of_players:
-            # display initial leaderboard
-            score_string, _ = await cringo.cringo_scoreboard(list_of_players)
+    async def send_emojis(self, emojis: List[List[str]]) -> None:
```

Comment:
`emojis` 🤔

Reply:
would you prefer emoji used as the plural?

Reply:
lol i'm just being a pedant, doesn't need to be changed